### PR TITLE
Support and Config Reconnect

### PR DESCRIPTION
**Pull request by betimer**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,8 +46,18 @@ module.exports = function(socket, opts) {
...
   function init() {
     socket.on('connect', function() {
-      queuedMessages.splice(0).forEach(bufferMessage);
-      signaller('connected');
+      if (!reconnected){
+        queuedMessages.splice(0).forEach(bufferMessage);
+        signaller('connected');
+      }
+      else {
+        console.log('pass as it is reconnect');
```
Comment (DamonOehlman) [inline index.js]: I think I'd remove the `else` statement here and the `console.log` as I don't think it's going to give you much value beyond debugging the initial implementation.

**Review by DamonOehlman — CHANGES_REQUESTED** [review]
Hey @betimer - I'm not doing much active maintenance on the rtc.io modules anymore (most of that is handled by the Coviu team) but here's a quick bit of feedback.  I think once this is done though we should merge it in as you've brought the package back up to date from a dependencies perspective.

I should probably do some reading about the `reconnect` event in socket.io and when it fires though...

**Comment by DamonOehlman** [comment]
> I should probably do some reading about the reconnect event in socket.io and when it fires though...

OK, so `reconnect` fires, but `connect` fires also. I wonder if it's better to implement the code such that you use a `once` listener for the `connect` and then `on` listeners for the `reconnect` calls.  Though, I'm not sure that you really need separate reconnect vs initial connect handling.

For instance, I'm not sure that you want to suppress sending buffered messages in the case of a reconnect (but I'm pretty rusty on all of this).  Assuming that a connection drops and messages are buffered during that disconnected time they would never be sent with your updates to the logic.

Honestly, after looking at the code I originally wrote here I really wish I had called the `bufferMessage` function `sendMessage` as it's what it's really doing (only buffering if it needs to).

**Comment by betimer** [comment]
Thanks @DamonOehlman 
I think putting a `once` makes more sense at this stage, I've made the update. Ye, the event will fire reconnect then connect each time the client tries to reconnect.

File: index.js (hunk 1 above)
Comment (betimer) [reply index.js]: Agree. It's only when debugging purpose.

**Comment by betimer** [comment]
But sorry I still not able to fix the travis error

**Comment by DamonOehlman** [comment]
> But sorry I still not able to fix the travis error

Maybe try bumping the `travis-multirunner` devdep to `4.5.0`?  The `3.x` is talking to an old version of the hosted service that reports current browser versions.  I'm not certain, but I'm pretty sure that will fix it.

**Comment by betimer** [comment]
@DamonOehlman Sorry still not able to fix the travis error, since the change does not impact the travis, can you help merge the pull request? so that we can manage a single source of truth.

Thank you

Xin

**Comment by DamonOehlman** [comment]
@betimer I'm comfortable with that.  I think the failures are flakey anyway, so I should look into that at some stage.

**Review by DamonOehlman — APPROVED** [review]
Going to merge this in and publish a new minor version (`2.3.0`)

**Comment by DamonOehlman** [comment]
@betimer done

**Comment by betimer** [comment]
@DamonOehlman Thank you very much

**Pull request merged**